src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -41,14 +39,14 @@
  */
 class MongoStatement implements StatementAdapter {
 
-    private final MongoClient mongoClient;
+    private final MongoDatabase mongoDatabase;
     private final MongoConnection mongoConnection;
     private final ClientSession clientSession;
 
     private boolean closed;
 
-    public MongoStatement(MongoClient mongoClient, ClientSession clientSession, MongoConnection mongoConnection) {
-        this.mongoClient = mongoClient;
+    public MongoStatement(MongoDatabase mongoDatabase, ClientSession clientSession, MongoConnection mongoConnection) {
+        this.mongoDatabase = mongoDatabase;
```

Comment:
We could instead keep requiring `mongoClient`, then get `mongoDatabase` as `mongoClient.getDatabase(mongoConnection.getSchema())`, but requiring `mongoDatabase` directly seems better to me.